ODIN_II/SRC/verilog_preprocessor.cpp

```diff
@@ -322,370 +389,324 @@ FILE* veri_preproc(FILE *source)
 
 	veri_preproc_bootstraped(source, preproc_producer, veri_initial);
 	rewind(preproc_producer);
+	
 	return preproc_producer;
 }
 
-/*
- * Returns a new temporary file with the comments removed.
- * Preserves the line numbers by keeping newlines in place.
- */
-FILE *remove_comments(FILE *source)
-{
-	FILE *destination = tmpfile();
-	destination = freopen(NULL, "r+", destination);
-	rewind(source);
-
-	char *line = NULL;
-	long line_size = 0;
-	int in_multiline_comment = FALSE;
-	while ((line = get_line(line, &line_size, source)) != nullptr)
-	{
-		unsigned int i;
-		for (i = 0; i < strnlen(line, line_size); i++)
-		{
-			if (!in_multiline_comment)
-			{
-				// For a single line comment, skip the rest of the line.
-				if (line[i] == '/' && line[i+1] == '/')
-				{
-					break;
-				}
-				// For a multi-line comment, set the flag and skip over the *.
-				else if (line[i] == '/' && line[i+1] == '*')
-				{
-					i++; // Skip the *.
-					in_multiline_comment = TRUE;
-				}
-				else
-				{
-					if (line[i] != '\n')
-						fputc(line[i], destination);
-				}
-			}
-			else
-			{
-				// If we're in a multi-line comment, search for the */
-				if (line[i] == '*' && line[i+1] == '/')
-				{
-					i++; // Skip the /
-					in_multiline_comment = FALSE;
-				}
-			}
-		}
-		fputc('\n', destination);
-		vtr::free(line);
-	}
-	rewind(destination);
-	return destination;
-}
 
 void veri_preproc_bootstraped(FILE *original_source, FILE *preproc_producer, veri_include *current_include)
 {
-	// Strip the comments from the source file producing a temporary source file.
-	FILE *source = remove_comments(original_source);
+	rewind(original_source);
 
 	int line_number = 1;
 	veri_flag_stack *skip = (veri_flag_stack *)vtr::calloc(1, sizeof(veri_flag_stack));;
```

Comment:
might as well take off the second semicolon here?

Reply:
should be fixed now :)